lib/ecto_job/worker.ex

```diff
@@ -47,9 +47,11 @@ defmodule EctoJob.Worker do
       queue.perform(JobQueue.initial_multi(job), job.params)
     rescue
       e ->
+        stacktrace = System.stacktrace()
+
         JobQueue.update_job_to_retrying(repo, job, DateTime.utc_now(), timeout)
 
-        reraise(e, System.stacktrace())
+        reraise(e, stacktrace)
```

Comment:
Wasn't there something in the latest version of Elixir that dealt with stack traces?

Reply:
you mean `__STACKTRACE__`? our version of elixir doesn't have that stuff yet. Though, I think that does the same as `System.stacktrace()`. 

As I understand it the problem is that the stacktrace is global var somewhere which `JobQueue.update_job_to_retrying(` chages it and binding it before the database call fixes it.